### ISSUE TITLE
Failed to compile iOS app

### ISSUE THREAD

**Issue by enahum** [Tencent/MMKV#444]
### The language of MMKV
Objective-C

### The version of MMKV
1.1.0

### The platform of MMKV
iOS

### The installation of MMKV
Cocoapods

### What's the issue?
Compiling the app to be run in a Simulator works fine but when compiling for release or to be run in a device I get the following errors

 /Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/ios/Pods/MMKVCore/Core/MMKV_OSX.cpp:66:9: use of undeclared identifier 'CRC32'

        CRC32 = mmkv::armv8_crc32;
                            ^~~~~~~~~~~~~~~

❌  /Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/ios/Pods/MMKVCore/Core/MMKV_OSX.cpp:66:23: no member named 'armv8_crc32' in namespace 'mmkv'

        CRC32 = mmkv::armv8_crc32;
        ^

**Comment by lingol** [comment]
Did you try to run in on WatchOS?

**Comment by enahum** [comment]
@lingol not at all, I'm trying to run it on an iPhone. Also having the same issues trying to build with fastlane

**Comment by lingol** [comment]
Can you try the `dev` branch? I believe that's been solved.

**Comment by lingol** [comment]
Also, the detail compile log will be helpful. It's something like this:
![image](https://user-images.githubusercontent.com/2088859/79089224-fbf5ed80-7d77-11ea-8ba6-602a400b375d.png)

**Comment by enahum** [comment]
@lingol  The dev branch has the same issue and this is what I could get from the logs

```
CompileC /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MemoryFile.o mattermost-mobile/ios/Pods/MMKVCore/Core/MemoryFile.cpp normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'MMKVCore' from project 'Pods')
    cd mattermost-mobile/ios/Pods
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target arm64-apple-ios8.0 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu++1z -stdlib=libc++ -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/enahum/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/enahum/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -Wno-c++11-extensions -DPOD_CONFIGURATION_DEBUG=1 -DDEBUG=1 -DCOCOAPODS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.4.sdk -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Winfinite-recursion -Wmove -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wrange-loop-analysis -Wno-semicolon-before-method-body -Wunguarded-availability -fembed-bitcode -iquote /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-generated-files.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-own-target-headers.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-all-non-framework-target-headers.hmap -ivfsoverlay /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/all-product-headers.yaml -iquote /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-project-headers.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/BuildProductsPath/Debug-iphoneos/MMKVCore/include -Imattermost-mobile/ios/Pods/Headers/Private -Imattermost-mobile/ios/Pods/Headers/Private/MMKVCore -Imattermost-mobile/ios/Pods/Headers/Public -Imattermost-mobile/ios/Pods/Headers/Public/MMKVCore -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources-normal/arm64 -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources/arm64 -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources -F/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/BuildProductsPath/Debug-iphoneos/MMKVCore -x objective-c++ -include mattermost-mobile/ios/Pods/Target\ Support\ Files/MMKVCore/MMKVCore-prefix.pch -MMD -MT dependencies -MF /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MemoryFile.d --serialize-diagnostics /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MemoryFile.dia -c mattermost-mobile/ios/Pods/MMKVCore/Core/MemoryFile.cpp -o /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MemoryFile.o

CompileC /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MMKV_OSX.o mattermost-mobile/ios/Pods/MMKVCore/Core/MMKV_OSX.cpp normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'MMKVCore' from project 'Pods')
    cd mattermost-mobile/ios/Pods
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target arm64-apple-ios8.0 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu++1z -stdlib=libc++ -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/enahum/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/enahum/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -Wno-c++11-extensions -DPOD_CONFIGURATION_DEBUG=1 -DDEBUG=1 -DCOCOAPODS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.4.sdk -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Winfinite-recursion -Wmove -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wrange-loop-analysis -Wno-semicolon-before-method-body -Wunguarded-availability -fembed-bitcode -iquote /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-generated-files.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-own-target-headers.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-all-non-framework-target-headers.hmap -ivfsoverlay /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/all-product-headers.yaml -iquote /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-project-headers.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/BuildProductsPath/Debug-iphoneos/MMKVCore/include -Imattermost-mobile/ios/Pods/Headers/Private -Imattermost-mobile/ios/Pods/Headers/Private/MMKVCore -Imattermost-mobile/ios/Pods/Headers/Public -Imattermost-mobile/ios/Pods/Headers/Public/MMKVCore -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources-normal/arm64 -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources/arm64 -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources -F/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/BuildProductsPath/Debug-iphoneos/MMKVCore -x objective-c++ -fno-objc-arc -include mattermost-mobile/ios/Pods/Target\ Support\ Files/MMKVCore/MMKVCore-prefix.pch -MMD -MT dependencies -MF /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MMKV_OSX.d --serialize-diagnostics /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MMKV_OSX.dia -c mattermost-mobile/ios/Pods/MMKVCore/Core/MMKV_OSX.cpp -o /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MMKV_OSX.o
mattermost-mobile/ios/Pods/MMKVCore/Core/MMKV_OSX.cpp:66:9: error: use of undeclared identifier 'CRC32'
        CRC32 = mmkv::armv8_crc32;
        ^
mattermost-mobile/ios/Pods/MMKVCore/Core/MMKV_OSX.cpp:66:23: error: no member named 'armv8_crc32' in namespace 'mmkv'
        CRC32 = mmkv::armv8_crc32;
                ~~~~~~^
2 errors generated.

CompileC /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MMKV_Android.o mattermost-mobile/ios/Pods/MMKVCore/Core/MMKV_Android.cpp normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'MMKVCore' from project 'Pods')
    cd mattermost-mobile/ios/Pods
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target arm64-apple-ios8.0 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu++1z -stdlib=libc++ -fobjc-arc -fobjc-weak -fmodules -fmodules-cache-path=/Users/enahum/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/enahum/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -Wno-c++11-extensions -DPOD_CONFIGURATION_DEBUG=1 -DDEBUG=1 -DCOCOAPODS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.4.sdk -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Winfinite-recursion -Wmove -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wrange-loop-analysis -Wno-semicolon-before-method-body -Wunguarded-availability -fembed-bitcode -iquote /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-generated-files.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-own-target-headers.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-all-non-framework-target-headers.hmap -ivfsoverlay /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/all-product-headers.yaml -iquote /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/MMKVCore-project-headers.hmap -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/BuildProductsPath/Debug-iphoneos/MMKVCore/include -Imattermost-mobile/ios/Pods/Headers/Private -Imattermost-mobile/ios/Pods/Headers/Private/MMKVCore -Imattermost-mobile/ios/Pods/Headers/Public -Imattermost-mobile/ios/Pods/Headers/Public/MMKVCore -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources-normal/arm64 -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources/arm64 -I/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/DerivedSources -F/Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/BuildProductsPath/Debug-iphoneos/MMKVCore -x objective-c++ -fno-objc-arc -include mattermost-mobile/ios/Pods/Target\ Support\ Files/MMKVCore/MMKVCore-prefix.pch -MMD -MT dependencies -MF /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MMKV_Android.d --serialize-diagnostics /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MMKV_Android.dia -c mattermost-mobile/ios/Pods/MMKVCore/Core/MMKV_Android.cpp -o /Users/enahum/Library/Developer/Xcode/DerivedData/Mattermost-czlinsdviifujheezzjvmisotjrm/Build/Intermediates.noindex/ArchiveIntermediates/Mattermost/IntermediateBuildFilesPath/Pods.build/Debug-iphoneos/MMKVCore.build/Objects-normal/arm64/MMKV_Android.o
warning: Could not read serialized diagnostics file: Cannot Load File: Failed to open diagnostics file (in target 'MMKVCore' from project 'Pods')
```

**Comment by enahum** [comment]
here is a screenshot from Xcode
![image](https://user-images.githubusercontent.com/6757047/79090609-3cbe1a00-7d18-11ea-9543-79a9cf55d09e.png)

**Comment by lingol** [comment]
That's weird. Everything looks fine, yet it didn't compile. 
I can't reproduce it locally. Would you provide a demo project that has been `pod install`? guoling@tencent.com.

**Comment by enahum** [comment]
@lingol you can use this https://github.com/mattermost/mattermost-mobile/pull/4145 this is a PR for the project we are using (it is build with React Native) the pod dependency is provided by https://github.com/ammarahm-ed/react-native-mmkv-storage/

**Comment by enahum** [comment]
Not sure what those lines are for, but as soon as I commented them out the project is building and data is being stored.

**Comment by lingol** [comment]
Hey, I have tried your project with that PR. Things work fine. Something must be wrong with your local checkout/pod install.
![image](https://user-images.githubusercontent.com/2088859/79092352-3913ad00-7d83-11ea-9d1e-970a978612f8.png)

**Comment by enahum** [comment]
Interesting... let me try and build it in the CI and see the results

**Comment by lingol** [comment]
> Interesting... let me try and build it in the CI and see the results.
> 
> Lingol's Max is your actual device I presume.

That's right.

**Comment by lingol** [comment]
> Not sure what those lines are for, but as soon as I commented them out the project is building and data is being stored.

Those are for crc32 optimization on arm64 devices. Commenting them out will fallback on zlib's implementation. Which is fine, a bit slower maybe.

**Comment by enahum** [comment]
> Those are for crc32 optimization on arm64 devices. Commenting them out will fallback on zlib's implementation. Which is fine, a bit slower maybe.

got it, yes,.. it is noticeable.

Anyway, lets see what the results are from the CI and I'll let you know. In any case this is the running build https://app.circleci.com/pipelines/github/mattermost/mattermost-mobile/3752/workflows/8c35b3ae-82e5-4a74-af89-e535a02e7627/jobs/5014

**Comment by enahum** [comment]
indeed it built correctly, it must be something else that was added that is causing some sort of conflict.

Thanks for all your help. will close the issue for now.

**Comment by huapeng222** [comment]
I have the same problem now.

**Comment by kyle-ssg** [comment]
Getting the same issue, is anyone else able to clarify if they can run on a device / archive without commenting out 

```
  CRC32 = mmkv::armv8_crc32;
```

Commenting this out does work but wouldn't be an option for me as I install pods in CI.

**Comment by lingol** [comment]
> Getting the same issue, is anyone else able to clarify if they can run on a device / archive without commenting out
> 
> ```
>   CRC32 = mmkv::armv8_crc32;
> ```
> 
> Commenting this out does work but wouldn't be an option for me as I install pods in CI.
> 
> ![image](https://user-images.githubusercontent.com/8608314/85945061-14b37f80-b933-11ea-8d3f-fe892e2187b7.png)
> 
> @lingol is there any reason why this would be the case? This is not only failing on my local environment but in CI too.

Consider using **New Build** instead of **Legacy Build**.

**Comment by kyle-ssg** [comment]
Hey, I'm using New Build for both CI and local

**Comment by lingol** [comment]
I've made a hotfix branch for you. Can you try that out? @kyle-ssg 
```
pod 'MMKV', :git => 'https://github.com/Tencent/MMKV.git', :branch => 'hotfix_xcode_build_error'
pod 'MMKVCore', :git => 'https://github.com/Tencent/MMKV.git', :branch => 'hotfix_xcode_build_error'
```

**Comment by kyle-ssg** [comment]
@lingol I can confirm this fixed the build

**Comment by lingol** [comment]
> @lingol I can confirm this fixed the build

Okay, a new version will come out this week.

**Comment by kyle-ssg** [comment]
Amazing, thanks so much!

**Comment by lingol** [comment]
Fixed in [v1.2.0](https://github.com/Tencent/MMKV/releases/tag/v1.2.0).
